diff --git a/app/models/host/managed.rb b/app/models/host/managed.rb
index 646f4a0..ca7192b 100644
--- a/app/models/host/managed.rb
+++ b/app/models/host/managed.rb
@@ -290,7 +290,7 @@ class Host::Managed < Host::Base
   def import_facts(facts)
     # Facts come from 'existing' attributes/infrastructure. We skip triggering
     # the orchestration of this infrastructure when we create a host this way.
-    skip_orchestration! if SETTINGS[:unattended]
+    skip_orchestration! if SETTINGS[:unattended] and not SETTINGS[:unattended_orchestration]
     super(facts)
   ensure
     enable_orchestration! if SETTINGS[:unattended]
diff --git a/app/models/setting/provisioning.rb b/app/models/setting/provisioning.rb
index 263d092..d40d002 100644
--- a/app/models/setting/provisioning.rb
+++ b/app/models/setting/provisioning.rb
@@ -29,6 +29,7 @@ class Setting::Provisioning < Setting
       self.set('access_unattended_without_build', N_("Allow access to unattended URLs without build mode being used"), false, N_('Access unattended without build')),
       self.set('manage_puppetca', N_("Foreman will automate certificate signing upon provision of new host"), true, N_('Manage PuppetCA')),
       self.set('ignore_puppet_facts_for_provisioning', N_("Stop updating IP address and MAC values from Puppet facts (affects all interfaces)"), false, N_('Ignore Puppet facts for provisioning')),
+      self.set('unattended_orchestration', N_("Enable host orchestration during unattended operations"), false, N_('Enable unattended orchestration')),
       self.set('ignored_interface_identifiers', N_("Ignore interfaces that match these values during facts importing, you can use * wildcard to match names with indexes e.g. macvtap*"), ['lo', 'usb*', 'vnet*', 'macvtap*', '_vdsmdummy_', 'veth*', 'docker*'], N_('Ignore interfaces with matching identifier')),
       self.set('ignore_facts_for_operatingsystem', N_("Stop updating Operating System from facts"), false, N_('Ignore facts for operating system')),
       self.set('ignore_facts_for_domain', N_("Stop updating domain values from facts"), false, N_('Ignore facts for domain')),
